src/helpers/useValidateUserPermissions.ts

```diff
@@ -0,0 +1,44 @@
+import { useSuspenseQuery } from "@tanstack/react-query";
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { getApiUrl } from './utils';
+
+export interface PermissionValidationRequest {
+    action: string;
+    object?: string;
+    scope?: string;
+}
+
+export interface PermissionValidationResponse extends PermissionValidationRequest{
+    allowed: boolean;
+}
+
+const validateUserPermissions = async (validations: PermissionValidationRequest[]): Promise<PermissionValidationResponse[]> => {
+  const { data } = await getAuthenticatedHttpClient().post(getApiUrl(`/api/authz/v1/permissions/validate/me`), validations);
```

Comment:
Nit: we often use a pattern where the actual API request lives in a `data/api.js` file.  (It's also used in this PR, elsewhere.)  Maybe we should follow it here, too?